src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/New-AzConnectedKubernetes.ps1

```diff
@@ -606,13 +577,77 @@ function New-AzConnectedKubernetes {
         else {
             $PSBoundParameters.Add('AgentPublicKeyCertificate', $AgentPublicKey)
         }
+
+        # Process the Arc agentry settings and protected settings
+        # Create any empty array of IArcAgentryConfigurations.
+        # shortened name to avoid class with type name.
+        #
+        # **NOTE** The Swagger naming does NOT match the names that will be used
+        #          in the final helm values file.  Instead there needs to be an
+        #          explicit mapping which is done in TWO places:
+        #          1. The ConfigDP is able to map the (unprotected) settings but
+        #             does not have access to the protected settings so...
+        #          2. This Powershell script has to perform the mapping for
+        #             protected settings.
+        #
+        #          This DOES mean that code changes are required both in the
+        #          Config DP annd this Powershell script if a new Kubernetes
+        #          feature is added.
+        # !!PDS: This may be wrong - should this be a hash of hashes?  no, it;s a list of hashes amnd hashes!
+        $arcAgentryConfigs = New-Object System.Collections.Generic.List[Microsoft.Azure.PowerShell.Cmdlets.ConnectedKubernetes.Models.Api20240715Preview.ArcAgentryConfigurations]
+
+        # Do not send protected settings to CCRP
+        $combinedKeys = $ConfigurationSetting.Keys + $RedactedProtectedConfiguration.Keys
+        $combinedKeys = $combinedKeys | Get-Unique
+        foreach ($feature in $combinedKeys) {
+            $ArcAgentryConfiguration = [Microsoft.Azure.PowerShell.Cmdlets.ConnectedKubernetes.Models.Api20240715Preview.ArcAgentryConfigurations]@{
+                Feature = $feature
+                Setting = $ConfigurationSetting[$feature]
```

Comment:
How can this work?  The fact that you are having to combine the keys implies that the redacted keys are not in `ConfigurationSettings` so shouldn't line 605 fail for those that are redacted?

Reply:
If the feature does not existing in `ConfigurationSettings` but is redacted, it means the feature exist only with protected settings. In this case, the arc agentry config will only contain feature with empty settings.